pkg/services/apiserver/service.go

```diff
@@ -210,6 +218,11 @@ func ProvideService(
 	s.rr.Group("/openapi", proxyHandler)
 	s.rr.Group("/version", proxyHandler)
 
+	if restConfig == nil {
+		restConfig = s
+		close(ready)
```

Comment:
oh interesting, i was thinking the service would call the function, but i like what you have better because i think you can remove the `ready` channel. the function [here](https://github.com/grafana/grafana/blob/dccca0729f9bfae7b4de2bd6dd0055fcb4998f8f/pkg/services/apiserver/service.go#L216C19-L216C24) uses AwaitRunning from dskit service, so it should block until the config is set

Reply:
scratch that, i see why it's needed

Reply:
it'd probably be useful for others to add a couple small code comments here about the duplicate `GetRestConfig` use case and what this pattern doing